framework/src/controller/bus.ts

```diff
@@ -195,26 +191,45 @@ export class Bus {
 		}
 	}
 
-	public async invoke<T>(actionData: string | JSONRPC.RequestObject): Promise<T> {
-		const action = Action.fromJSONRPC(actionData);
-		const actionFullName = action.key();
+	public async invoke<T>(
+		actionData: string | JSONRPC.RequestObject,
+	): Promise<JSONRPC.ResponseObjectWithResult<T>> {
+		const parsedAction = Action.fromJSONRPCRequest(actionData);
+
+		try {
+			JSONRPC.validateJSONRPCRequest(parsedAction.toJSONRPCRequest() as never);
+		} catch (error) {
+			this.logger.error(
+				JSONRPC.errorResponse(parsedAction.id, JSONRPC.invalidRequest()),
```

Comment:
the `JSONRPC.invalidRequest` already convey its invalid request. so maybe the data can be about the validation error which tells about what property of json rpc request is incorrect?

Reply:
The error handling is improved in upcoming PR.